src/main/java/seedu/address/model/student/UniqueStudentList.java

```diff
@@ -145,30 +146,74 @@ public ObservableList<Student> getInternalList() {
      * @param tr a timeRange object containing the {@code startTime}, {@code endTime} and {@code duration}.
      * @return the next first available Class.
      */
-    public Class findAvailableClass(TimeRange tr) {
+    public Class findAvailableClass(TimeRange tr, LocalTime currTime) {
         LocalDate currDate = LocalDate.now();
-        List<Student> list = internalList
+        List<Class> listAfterToday = internalList
                 .stream()
                 .filter(student -> student.getAClass().startTime != null
                         && student.getAClass().endTime != null
                         && student.getAClass().date != null
-                        && student.getAClass().date.compareTo(currDate) >= 0
-                        && student.getAClass().startTime.compareTo(LocalTime.now()) >= 0)
-                .sorted(Student::compareToByClassStartTimeAsc)
+                        && student.getAClass().date.compareTo(currDate) > 0)
+                .sorted(Student::compareToByClassAsc)
+                .map((element) -> element.getAClass())
                 .collect(Collectors.toList());
-        Class newClass = new Class();
+        List<Class> listSameDay = internalList
+                .stream()
+                .filter(student -> student.getDisplayedClass().startTime != null
+                        && student.getDisplayedClass().endTime != null
+                        && student.getDisplayedClass().date != null
+                        && student.getDisplayedClass().date.compareTo(currDate) == 0)
+                .sorted(Student::compareToByClassAsc)
+                .map((element) -> element.getDisplayedClass())
```

Comment:
After the merge of Caden's PR to remove `displayClass` field, this part needs to be changed accordingly.